includes/class-events-store-cron-filters.php

```diff
@@ -0,0 +1,241 @@
+<?php
+/**
+ * Event storage for WP 5.1 and later.
+ *
+ * @package a8c_Cron_Control
+ */
+
+namespace Automattic\WP\Cron_Control;
+
+/**
+ * Class Option_Intercept
+ */
+class Events_Store_Cron_Filters extends Singleton {
+	/**
+	 * Register hooks to intercept events before storage.
+	 */
+	protected function class_init() {
+		add_filter( 'pre_schedule_event', [ $this, 'filter_event_scheduling' ], 10, 2 );
+		add_filter( 'pre_reschedule_event', [ $this, 'filter_event_rescheduling' ], 10, 2 );
+		add_filter( 'pre_unschedule_event', [ $this, 'filter_event_unscheduling' ], 10, 4 );
```

Comment:
@ethitter What do you think about naming the callbacks with the filter name? So `function filter_pre_unschedule_event()` for example. That would make it easier to know when they are firing at a glance without referring back here.

Reply:
That's fine with me. I believe that in an earlier iteration, some of the callbacks were shared between filters, but that's no longer the case.